src/Caching/Memory/test/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -121,6 +122,34 @@ public async Task GetOrCreateAsync_AddsNewValue()
             Assert.Same(obj, result);
         }
 
+        [Fact]
+        public async Task GetOrCreateAsync_DoesNotBlock()
+        {
+            var singleThreadedContext = new SingleThreadSynchronizationContext();
+            var singleThreadedScheduler = new SynchronizationContextTaskScheduler(singleThreadedContext);
+            var synhronizationContextTask = Task.Run(() => singleThreadedContext.RunOnCurrentThread());
+
+            var cache = CreateCache();
+
+            object SystemUnderTestMethod()
+            {
+                var result = cache.GetOrCreateAsync("key", async e =>
+                {
+                    await Task.Delay(1).ConfigureAwait(false);
+                    return new object();
+                }).GetAwaiter().GetResult();
+                return result;
+            }
+
+            var sut = Task.Factory.StartNew(SystemUnderTestMethod, CancellationToken.None, TaskCreationOptions.None, singleThreadedScheduler);
+
+            var exception = await Record.ExceptionAsync(() => sut.TimeoutAfter(TimeSpan.FromSeconds(3)));
```

Comment:
What's the purpose of recording the exception just to check it's `null`? I'd think `await sut.TimeoutAfter(...)` would be sufficient because if it throws, the test will fail and will display the exception details.

Reply:
Got it, just being explicit on what is being asserted. Will update as suggested.